src/nm.c

```diff
@@ -515,297 +504,287 @@ write_nm_conf_access_point(NetplanNetDefinition* def, const char* rootdir, const
 
-        if (def->bridge_params.path_cost || def->bridge_params.port_priority)
-            g_string_append_printf(s, "\n[bridge-port]\n");
         if (def->bridge_params.path_cost)
-            g_string_append_printf(s, "path-cost=%u\n", def->bridge_params.path_cost);
+            g_key_file_set_uint64(kf, "bridge-port", "path-cost", def->bridge_params.path_cost);
         if (def->bridge_params.port_priority)
-            g_string_append_printf(s, "priority=%u\n", def->bridge_params.port_priority);
+            g_key_file_set_uint64(kf, "bridge-port", "priority", def->bridge_params.port_priority);
+    }
+    if (def->bond) {
+        g_key_file_set_string(kf, "connection", "slave-type", "bond");
+        g_key_file_set_string(kf, "connection", "master", def->bond);
     }
-    if (def->bond)
-        g_string_append_printf(s, "slave-type=bond\nmaster=%s\n", def->bond);
 
     if (def->ipv6_mtubytes) {
         g_fprintf(stderr, "ERROR: %s: NetworkManager definitions do not support ipv6-mtu\n", def->id);
         exit(1);
     }
 
     if (def->type < NETPLAN_DEF_TYPE_VIRTUAL) {
-        GString *link_str = NULL;
-
-        link_str = g_string_new(NULL);
+        g_key_file_set_integer(kf, "ethernet", "wake-on-lan", def->wake_on_lan ? 1 : 0);
 
-        g_string_append_printf(s, "\n[ethernet]\nwake-on-lan=%i\n", def->wake_on_lan ? 1 : 0);
-
-        if (!def->set_name && def->match.mac) {
-            g_string_append_printf(link_str, "mac-address=%s\n", def->match.mac);
-        }
-        if (def->set_mac) {
-            g_string_append_printf(link_str, "cloned-mac-address=%s\n", def->set_mac);
-        }
-        if (def->mtubytes) {
-            g_string_append_printf(link_str, "mtu=%u\n", def->mtubytes);
-        }
-        if (def->wowlan && def->wowlan > NETPLAN_WIFI_WOWLAN_DEFAULT)
-            g_string_append_printf(link_str, "wake-on-wlan=%u\n", def->wowlan);
-
-        if (link_str->len > 0) {
-            switch (def->type) {
-                case NETPLAN_DEF_TYPE_WIFI:
-                    g_string_append_printf(s, "\n[802-11-wireless]\n%s", link_str->str);  break;
-                case NETPLAN_DEF_TYPE_MODEM:
-                    /* Avoid adding an [ethernet] section into the [gsm/cdma] description. */
-                    break;
-                default:
-                    g_string_append_printf(s, "\n[802-3-ethernet]\n%s", link_str->str);  break;
-            }
-        }
-
-        g_string_free(link_str, TRUE);
-    } else {
-        GString *link_str = NULL;
-
-        link_str = g_string_new(NULL);
-
-        if (def->set_mac) {
-            g_string_append_printf(link_str, "cloned-mac-address=%s\n", def->set_mac);
-        }
-        if (def->mtubytes) {
-            g_string_append_printf(link_str, "mtu=%u\n", def->mtubytes);
+        const char* con_type = NULL;
+        switch (def->type) {
+            case NETPLAN_DEF_TYPE_WIFI:
+                con_type = "wifi";
```

Comment:
Just making sure, since I'm not that into NM right now: is this really `wifi` right now? Since I remember `802-11-wireless` and the docs also mention the same: https://developer.gnome.org/NetworkManager/stable/settings-802-11-wireless.html

Is `wifi` an alias or something?

Reply:
Yes, I explicitly researched this as I wasn't sure and didn't change it in the first place. But I saw we had mixed definitions using 'wifi' and '802-11-wireless' (same for ethernet) and thought it makes sense to clean it up with this refactoring.

Both keywords are possible, but upstream NetworkManager switched over to using the short alias by default (wifi, wifi-security and ethernet) some 7 years ago. So I think we should do the same. See: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/c36200a225aefb2a3919618e75682646899b82c0